homeassistant/components/youtube/__init__.py

```diff
@@ -58,15 +58,17 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     return unload_ok
 
 
-async def delete_devices(
+async def delete_stale_devices(
     hass: HomeAssistant, entry: ConfigEntry, coordinator: YouTubeDataUpdateCoordinator
 ) -> None:
-    """Delete all devices created by integration."""
+    """Delete stale devices."""
     channel_ids = list(coordinator.data)
     device_registry = dr.async_get(hass)
     dev_entries = dr.async_entries_for_config_entry(device_registry, entry.entry_id)
     for dev_entry in dev_entries:
-        if any(identifier[1] in channel_ids for identifier in dev_entry.identifiers):
+        if not any(
```

Comment:
A test would be good.

Example:
https://github.com/home-assistant/core/blob/dec9eb9ae496d98c8fc38a5f680e25451d84790c/tests/components/tradfri/test_init.py#L54-L91